src/GraphQL.MicrosoftDI/GraphQLBuilderExtensions.cs

```diff
@@ -12,14 +12,15 @@ namespace GraphQL.MicrosoftDI
     public static class GraphQLBuilderExtensions
     {
         /// <summary>
-        /// Returns a new <see cref="IGraphQLBuilder"/> for the specified service collection and
+        /// Configures a GraphQL pipeline using the configuration delegate passed into
+        /// <paramref name="configure"/> for the specified service collection and
         /// registers a default set of services required by GraphQL if they have not already been registered.
         /// <br/><br/>
         /// Does not include <see cref="IDocumentWriter"/>, and the default <see cref="IDocumentExecuter"/>
         /// implementation does not support subscriptions.
         /// </summary>
-        public static IGraphQLBuilder AddGraphQL(this IServiceCollection services)
-            => new GraphQLBuilder(services);
+        public static void AddGraphQL(this IServiceCollection services, Action<IGraphQLBuilder> configure)
```

Comment:
I was not against returning `IGraphQLBuilder`. This will allow you to easily pass it in third-party code.

Reply:
`Action<IGraphQLBuilder>` ->  `Action<IGraphQLBuilder>?`

Reply:
You don't think it would encourage an 'incorrect' coding pattern?  Is that what other similar design patterns do?

Reply:
> This will allow you to easily pass it in third-party code.

Perhaps, but I'm having difficulty imagining such a scenario.  This is only an extension method that serves as an entry point to the configuration.  This extension method is provided by the package for the specific DI provider, not in the main GraphQL package.  If you wanted to write your own extension method, like for example, `AddSpecialGraphQL<TSchema>`, you'd probably be creating an instance of GraphQL.MicrosoftDI.GraphQLBuilder directly, likely with a custom delegate, rather than calling this one-line extension method.

For example:

```cs
public static void AddMySpecialGraphQL<TSchema>(this IServiceCollection serviceCollection, Action<IGraphQLBuilder>? configure = null)
{
    var assembly = Assembly.GetCallingAssembly();
    serviceCollection.AddGraphQL(c =>
    {
        c.AddSchema<TSchema>();
        c.AddSystemTextJson();
        c.AddGraphTypes(assembly);
        configure?.Invoke(c);
    });
}
```

Any extension methods that operate on IGraphQLBuilder belong inside the configuration delegate, not outside of it, with this new pattern, so code in RegisterDefaultServices will always execute last.

Finally, if it was necessary for someone to execute configuration actions after RegisterDefaultServices, they could capture the configuration variable into a local variable and execute it after AddGraphQL completes.  But I feel that it would be too easy to accidentally write configuration methods outside of the delegate if AddGraphQL returns the builder.

For example:

```cs
//for example if a user had this
services.AddGraphQL(config => config
    .AddSchema<MySchema>()
    .AddSystemTextJson());

//they could write this and not realize that it is 'incorrect'
services.AddGraphQL(config => config
    .AddSchema<MySchema>()
    .AddSystemTextJson())   // <-- extra parentheses from before
    .AddDataLoader(); // <-- missing parentheses
```

The second example still compiles, and is difficult to see the difference.

Reply:
Well... I understand all you said. I guess I wanted to make another thing and wrote badly. Let's return `IServiceCollection` to allow outer configuration chaining.

Reply:
Sorry.

Reply:
Good idea

Reply:
Done